registry/proxy/proxyregistry.go

```diff
@@ -117,8 +117,13 @@ func (pr *proxyingRegistry) Repositories(ctx context.Context, repos []string, la
 }
 
 func (pr *proxyingRegistry) Repository(ctx context.Context, name reference.Named) (distribution.Repository, error) {
+	hcm, ok := pr.authChallenger.(*remoteAuthChallenger)
```

Comment:
Not a fan of having a type assertion here. Could we extend the `authChallenger` interface to have getters for `ChallengeManager` and `CredentialStore`?